consensus/state.go

```diff
@@ -1033,31 +1074,69 @@ func (cs *ConsensusState) AddProposalBlockPart(height int, part *types.Part) (ad
 		log.Info("Received complete proposal", "hash", cs.ProposalBlock.Hash())
 		if cs.Step == RoundStepPropose && cs.isProposalComplete() {
 			// Move onto the next step
-			go cs.EnterPrevote(height, cs.Round)
+			go cs.EnterPrevote(height, cs.Round, false)
 		} else if cs.Step == RoundStepCommit {
 			// If we're waiting on the proposal block...
-			cs.tryFinalizeCommit(height)
+			cs.tryFinalizeCommit(height, cs.Round)
```

Comment:
We need both +2/3 precommits and the complete proposal from the commit round to finalize commit.  Sometimes we’ll have the +2/3 precommits first;  in that case AddProposalBlockPart() is what triggers the finalizecommit, but it needs to be on the commitRound.  But I think best is to just track commitRound in the RoundState and not pass it in as a parameter to (try)FinalizeCommit()